File: core/src/ru/myitschool/dcrawler/entities/Character.java
package ru.myitschool.dcrawler.entities;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.utils.Array;

/**
 * Created by Voyager on 25.04.2017.
 */
public class Character extends Entity {

    private static final int CHARS_MAX = 4;

    private boolean moved = false;
    private static boolean created = false;

    private static Array<Character> chars = new Array<>(CHARS_MAX);

    public Character(Texture texture, Texture portrait, float x, float y, int hp, int hpMax, int mp, int mpMax, int armor) {
        super(texture, portrait, x, y, hp, hpMax, mp, mpMax, armor);
        chars.add(this);
        Entity.add(this);
    }

    public static Character getChar(int num) {
        return chars.get(num);
    }

    public static Array<Character> getChars() {
        return chars;
    }

    public static boolean isCreated() {
        return created;
    }

    public static void setCreated(boolean created) {
        Character.created = created;
    }

    @Override
    public boolean isPlayer() {
        return true;
    }

    @Override
    public boolean turnIsEnded() {
        return false;
    }

    @Override
    public void endTurn() {
        super.endTurn();
    }

    @Override
    public void startTurn() {
        moved = false;
        setMp(getMpMax());
        super.startTurn();
    }

    @Override
    public void startMove() {
        super.startMove();
    }

    @Override
    public void endMove() {
        moved = true;
        super.endMove();
        setMp(0);
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/Character.kt
package com.wolg_vlad.dcrawler.entities

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.utils.Array

/**
 * Created by Voyager on 25.04.2017.
 */
class Character(texture: Texture, portrait: Texture, x: Float, y: Float, hp: Int, hpMax: Int, mp: Int, mpMax: Int, armor: Int) : Entity(texture, portrait, x, y, hp, hpMax, mp, mpMax, armor) {
    private var moved = false
    override val isCharacter: Boolean
        get() = true

    override fun turnIsEnded(): Boolean {
        return false
    }

    override fun endTurn() {
        super.endTurn()
    }

    override fun startTurn() {
        moved = false
        super.startTurn()
    }

    override fun startMove() {
        super.startMove()
    }

    override fun endMove() {
        moved = true
        super.endMove()
        setMp(0)
    }

    companion object {
        private const val CHARS_MAX = 4
        var created = false
        val chars = Array<Character>(CHARS_MAX)
        fun getChar(num: Int): Character {
            return chars[num]
        }

    }

    init {
        chars.add(this)
        add(this)
    }
}